### ISSUE TITLE
Error using mex - No supported compiler or SDK was found

### ISSUE THREAD

**Issue by rodrigovimieiro** [CERN/TIGRE#80]
## Expected Behavior
I'm trying to run Compile.m to run demo files after.

## Actual Behavior

Running Compile.m file and get this error:

Error using mex
No supported compiler or SDK was found. For options, visit https://www.mathworks.com/support/compilers.

Error in Compile (line 110)
        mex -largeArrayDims ./Source/Ax_mex.cpp ./Source/ray_interpolated_projection.cu
        ./Source/Siddon_projection.cu ./Source/ray_interpolated_projection_parallel.cu
        ./Source/Siddon_projection_parallel.cu -outdir ./Mex_files/win64

## Code to reproduce the problem (If applicable)

```

```

## Specifications 

  - MATLAB version: R2017b
  - OS: Win 10 x64
  - CUDA version: Cuda v9.2
  - Visual Studio 2015 (Also tried 2017) 
  -  MinGW64 Compiler (c)

**Comment by AnderBiguri** [comment]
Hi @rodrigovimieiro, 
Try to type these line in MATLAB and let me know what they say:

mex -setup

getenv('CUDA_PATH')

**Comment by rodrigovimieiro** [comment]
Hi Ander, thank you!

#############
mex -setup

MEX configured to use 'MinGW64 Compiler (C)' for C language compilation.
Warning: The MATLAB C and Fortran API has changed to support MATLAB
	 variables with more than 2^32-1 elements. You will be required
	 to update your code to utilize the new API.
	 You can find more information about this at:
	 http://www.mathworks.com/help/matlab/matlab_external/upgrading-mex-files-to-use-64-bit-api.html.

To choose a different language, select one from the following:
 mex -setup C++ 
 mex -setup FORTRAN
#############

getenv('CUDA_PATH')

ans =  'C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2'

**Comment by AnderBiguri** [comment]
@rodrigovimieiro  There is your problem. You are not compiling with MVS, you are compiling with MinGW64, which does not support CUDA.  Try `mex -setup C` or `mex -setup C++`. It shoudl give you the option of Visual Studio at some point. If it does not show, something is wrong with the installation I think

**Comment by rodrigovimieiro** [comment]
I installed MVS 2015, but it seems Matlab doesn't recognize it. 

When I installed MVS 2017, Matlab recognized but in your documentation it says that this version doesn't work. 

####
mex -setup
MEX configured to use 'Microsoft Visual C++ 2017 (C)' for C language compilation.
Warning: The MATLAB C and Fortran API has changed to support MATLAB
	 variables with more than 2^32-1 elements. You will be required
	 to update your code to utilize the new API.
	 You can find more information about this at:
	 http://www.mathworks.com/help/matlab/matlab_external/upgrading-mex-files-to-use-64-bit-api.html.

To choose a different language, select one from the following:
 mex -setup C++ 
 mex -setup FORTRAN

**Comment by AnderBiguri** [comment]
@rodrigovimieiro  I suggest you try it anyway. 

MVS 2015 has some setup missing then, as I have used it with MATLAB.

**Comment by rodrigovimieiro** [comment]
I think I have installed. I'm attaching an image of what's installed.

![img](https://user-images.githubusercontent.com/29632934/41107651-261d7324-6a49-11e8-98f1-8f6a7e77dcbe.png)

Btw, do you have the files already compiled?

**Comment by AnderBiguri** [comment]
@rodrigovimieiro You may have missed the comments in the post I linked. VisualStudio 2015 does not install C/C++ by default. Just make sure you have indeed selected specifically to install C/C++ in the  installation procedure, or you won't have it. 

Also, have you tried compiling with 2017?

**Comment by rodrigovimieiro** [comment]
@AnderBiguri I followed the instruction in the link you sent. 

Now, when I run the Compile.m code I get this:

-------------------
Compile
MEX configured to use 'Microsoft Visual C++ 2015 (C)' for C language compilation.
Warning: The MATLAB C and Fortran API has changed to support MATLAB
	 variables with more than 2^32-1 elements. You will be required
	 to update your code to utilize the new API.
	 You can find more information about this at:
	 http://www.mathworks.com/help/matlab/matlab_external/upgrading-mex-files-to-use-64-bit-api.html.

To choose a different language, select one from the following:
 mex -setup C++ 
 mex -setup FORTRAN
Compiling TIGRE source...
This may take a couple of minutes....
Error using mex
No supported compiler or SDK was found. For options, visit https://www.mathworks.com/support/compilers.

Error in Compile (line 110)
        mex -largeArrayDims ./Source/Ax_mex.cpp ./Source/ray_interpolated_projection.cu
        ./Source/Siddon_projection.cu ./Source/ray_interpolated_projection_parallel.cu
        ./Source/Siddon_projection_parallel.cu -outdir ./Mex_files/win64
-----------------
I tried with VS 2017 and I had the same problem.

**Comment by AnderBiguri** [comment]
@rodrigovimieiro Lets try the following: 

Open `mex_CUDA_win64.xml` and replace any instance of  `VS120COMNTOOLS` by `VS140COMNTOOLS`.

Apologies for all this, I am trying to make it compile easier.... Still work in progress.

**Comment by rodrigovimieiro** [comment]
@AnderBiguri It worked for a couple of minutes, but it got stuck in this error:

Compiling TIGRE source...
This may take a couple of minutes....
Building with 'NVIDIA CUDA Compiler'.
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(144): warning: variable "azm" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(145): warning: variable "azM" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(144): warning: variable "azm" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(145): warning: variable "azM" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(144): warning: variable "azm" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(145): warning: variable "azM" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(144): warning: variable "azm" was declared but never referenced
C:/Users/user1/Documents/Rodrigo/TIGRE-master/MATLAB/Source/Siddon_projection_parallel.cu(145): warning: variable "azM" was declared but never referenced

**Error using mex
LINK : fatal error LNK1104: cannot open file 'ucrt.lib'



Error in Compile (line 110)
        mex -largeArrayDims ./Source/Ax_mex.cpp ./Source/ray_interpolated_projection.cu
        ./Source/Siddon_projection.cu ./Source/ray_interpolated_projection_parallel.cu
        ./Source/Siddon_projection_parallel.cu -outdir ./Mex_files/win64**

**Comment by AnderBiguri** [comment]
@rodrigovimieiro  https://social.msdn.microsoft.com/Forums/en-US/7a2f61b7-3c01-4767-9a30-e897dc240e60/lnk1104-cannot-open-file-ucrtdlib?forum=vssetup This may be the solution.

**Comment by rodrigovimieiro** [comment]
@AnderBiguri 
Got it.

I installed Windows SDK for Windows 10 from the link you sent, but i didn't work.

So I found the solution here: https://www.mathworks.com/matlabcentral/answers/97597-why-do-i-receive-a-libc-lib-error-when-doing-a-mex-compilation-in-matlab-7-9-r2009b

I placed the missing files in the matlab working directory.

![img](https://user-images.githubusercontent.com/29632934/41112948-c0d5837c-6a56-11e8-93e4-301f399b5190.png)

I will run the demo codes now! Thank you!
